schema/fields.go

```diff
@@ -409,14 +410,15 @@ func (f *Field) IsSorted() bool {
 }
 
 func (f *Field) IsCompatible(f1 *Field) error {
-	if f.DataType != f1.DataType {
+	if f.DataType != f1.DataType && !config.DefaultConfig.Schema.AllowIncompatible {
 		return errors.InvalidArgument("data type mismatch for field %q", f.FieldName)
 	}
 
 	if f.IsPrimaryKey() != f1.IsPrimaryKey() {
 		return errors.InvalidArgument("primary key changes are not allowed %q", f.FieldName)
 	}
 
+	// TODO: Implement handling MaxLength reduction by schema change
```

Comment:
Do we have a ticket or something to track this so we know we need to come back to this? 
Or are you planning to work on this soon?